src/planner.cc

```diff
@@ -557,21 +631,174 @@ bool Planner::Impl::machine_move(const AxesRegister &axis, float feedrate) {
 
 void Planner::Impl::bring_path_to_halt() {
   if (path_halted_) return;
-  // Enqueue a new position that is the same position as the last
-  // one seen, but zero speed. That will allow the previous segment to
-  // slow down. Enqueue.
-  struct AxisTarget *previous = planning_buffer_.back();
-  struct AxisTarget *new_pos = planning_buffer_.append();
-  for (const GCodeParserAxis a : AllAxes()) {
-    new_pos->position_steps[a] = previous->position_steps[a];
-    new_pos->delta_steps[a] = 0;
+
+  // Flush the queue.
+  issue_motor_move_if_possible(true);
+}
+
+// Perform a backward and forward pass across
+// the whole planning buffer to adapt the start
+// and final speed of all the segments.
+void Planner::Impl::UpdateMotionProfile() {
+  assert(planning_buffer_.size());
```

Comment:
After a rebase, this can be `assert(!planning_buffer_.empty());`
(and clang-tidy will complain if it is not :) ).

Reply:
(solved)